telemetry/request_factory.go

```diff
@@ -97,24 +104,36 @@ func (f *requestFactory) buildRequest(entries []PayloadEntry, getPayloadBytes pa
 		}
 	}
 
-	buf := &bytes.Buffer{}
-	getPayloadBytes(buf, entries)
+	// Grab a buffer from the cached buffers and reset it
+	decompressedBuffer := configuredFactory.uncompressedBuffers.Get().(*bytes.Buffer)
+	decompressedBuffer.Reset()
+
+	// Grab a gzip structure (and buffer) from the cache and reset it
+	poolEntry := configuredFactory.zippers.Get().(*gzipPoolEntry)
```

Comment:
Is there any advantage to having the gzip pool entries also carry a buffer? Why not just get two byte buffers from the `uncompressedBuffers` pool?

Reply:
We want to reuse the buffer after the size has been adjusted by the gzip writer. Those buffers will bias to smaller sizes, therefore they are kept separate.

Reply:
I buy that.